src/ray/core_worker/core_worker.cc

```diff
@@ -1211,9 +1211,11 @@ Status CoreWorker::CancelTask(const ObjectID &object_id, bool force_kill) {
     return Status::Invalid("Actor task cancellation is not supported.");
   }
   rpc::Address obj_addr;
-  if (!reference_counter_->GetOwner(object_id, nullptr, &obj_addr) ||
-      obj_addr.SerializeAsString() != rpc_address_.SerializeAsString()) {
-    return Status::Invalid("Task is not locally submitted.");
+  if (!reference_counter_->GetOwner(object_id, nullptr, &obj_addr)) {
+    RAY_LOG(ERROR) << "No owner for object";
+  }
+  if (obj_addr.SerializeAsString() != rpc_address_.SerializeAsString()) {
```

Comment:
shouldn't this be `else if`? if we don't have the owner for the object, we probably don't even want to do this check

Reply:
actually see above, we should just return a non-OK status if we don't get the owner

Reply:
+1